pkg/controller/dgraph/models/container.go

```diff
@@ -86,6 +95,12 @@ func StoreAndRetrieveContainers(pod api_v1.Pod, podUID string) []*Container {
 
 		containers = append(containers, container)
 	}
+	if isDeleted {
```

Comment:
Can we split this into smaller functions? This function is handling too many responsibilities.